Code/GraphMol/MolDraw2D/DrawMol.cpp

```diff
@@ -953,6 +969,82 @@ void DrawMol::extractCloseContacts() {
   }
 }
 
+namespace {
+bool doesLabelClashWithBonds(
+    AtomSymbol &atsym1, const std::vector<std::unique_ptr<DrawShape>> &bonds) {
+  for (auto &rect1 : atsym1.rects_) {
+    auto oldTrans = rect1->trans_;
+    rect1->trans_ += atsym1.cds_;
+    for (const auto &bond : bonds) {
+      bool res = bond->doesRectClash(*rect1, 0.0);
+      if (res) {
+        rect1->trans_ = oldTrans;
+        return true;
+      }
+    }
+    rect1->trans_ = oldTrans;
+  }
+  return false;
+}
+
+bool doesLabelClashWithLabels(
+    unsigned idx, const std::vector<std::unique_ptr<AtomSymbol>> &atomLabels) {
+  for (size_t i = 0; i < atomLabels.size(); ++i) {
+    if (i == idx || atomLabels[i] == nullptr) {
+      continue;
+    }
+    if (doLabelsClash(*atomLabels[idx], *atomLabels[i])) {
+      return true;
+    }
+  }
+  return false;
+}
+}  // namespace
+
+// ****************************************************************************
+void DrawMol::resolveAtomSymbolClashes() {
+  const static std::vector<OrientType> allOrients{OrientType::N, OrientType::E,
+                                                  OrientType::S, OrientType::W};
+  for (auto at1 : drawMol_->atoms()) {
+    const auto atIdx1 = at1->getIdx();
+    for (auto at2 : drawMol_->atoms()) {
+      const auto atIdx2 = at2->getIdx();
+      if (atIdx1 == atIdx2) {
```

Comment:
Wouldn't this be more efficient (yet still correct) as:
```suggestion
      if (atIdx1 >= atIdx2) {
```
We don't need to compare rect A against rect B and rect B against A, right?

Reply:
I think it’s worse than that. A late “refinement” has it only moving one of the 2 whichever order they’re in. I will re-work it.